dev_guide/builds.adoc

```diff
@@ -818,32 +817,72 @@ endif::[]
 
 // end::using-a-proxy-for-git-cloning-1[]
 
-[[trusted-certificate-authorities]]
-==== Trusted Certificate Authorities
+[[gitconfig-file]]
+==== Gitconfig File
 
-The set of TLS certificate authorities that are trusted during a `git clone`
-operation are built into the {product-title} infrastructure images.  If your Git
-server uses a self-signed certificate or one signed by an authority not trusted
-by the image, you can disable TLS  verification by setting the
-`*GIT_SSL_NO_VERIFY*` environment variable to *true* in the appropriate strategy
-section of your build configuration. You can use the
-xref:../dev_guide/environment_variables.adoc#overview[`oc set env`] command to
-manage `*BuildConfig*` environment variables.
+If the cloning of your application is dependent on a `.gitconfig` file
+you can create a secret that contains it, and then add
+it to the builder service account, and then your `BuildConfig`.
 
-[[using-private-repositories-for-builds]]
-==== Using Private Repositories for Builds
+. Create desired type of `*secret*`:
 
-Supply valid credentials to build an application from a private repository.
+.. To create a secret from a `.gitconfig`:
++
+====
+----
+$ oc secrets new mysecret .gitconfig=path/to/.gitconfig
+----
+====
++
+[NOTE]
+====
+SSL verification can be turned off, if `sslVerify=false` is set for the `http`
+section in your `.gitconfig` file:
+----
+[http]
+        sslVerify=false
+----
+====
 
-Currently two types of authentication are supported: basic username-password
-and SSH key based authentication.
+.  Add the `*secret*` to the builder service account. Each build is run with the
+the `builder` role, so you need to give it access your secret with the
+following command:
++
+====
+----
+$ oc secrets link builder mysecret
+----
+====
+
+.  Add the `*secret*` to the `*BuildConfig*`:
++
+====
+----
+source:
+  git:
+    uri: "https://github.com/openshift/nodejs-ex.git"
+  sourceSecret:
+    name: "mysecret"
+----
+====
++
+You can also use the `*oc set build-secret*` command to set the secret on the
+existing build configuration:
++
+====
+----
+$ oc set build-secret --source bc/sample-build mysecret
+----
+====
+
+xref:builds.adoc#using-secrets-in-the-buildconfig[Defining Secrets in the
+BuildConfig] provides more information on this topic.
```

Comment:
this content (add secret to service account, reference secret from buildconfig, use oc set) is now redundant in each of these sections.  I think we can do better by giving an intro to the concept of buildconfigs having source secrets, and then go into the details of the various things (gitconfig, auth, etc) you might put in those secrets.